base/compiler/abstractinterpretation.jl

```diff
@@ -1548,6 +1556,12 @@ function abstract_apply(interp::AbstractInterpreter, argtypes::Vector{Any}, si::
                 end
                 cti = Any[Vararg{argt}]
             end
+            effects = merge_effects(effects, ai_effects)
+            if info !== nothing
```

Comment:
```suggestion
            # merge effects of the `iterate` call
            effects = merge_effects(effects, ai_effects)
            # merge effects of call(s) with the iterated arguments
            if info !== nothing
```

Reply:
It's not really the effects of the iterate call. It's more the effects of imprecision of the iterate call. The `info.each`, are all methods of `iterate`.

Reply:
Ah, I understand.